be/src/vec/exec/format/json/new_json_reader.cpp

```diff
@@ -1007,4 +1015,560 @@ Status NewJsonReader::_read_one_message(std::unique_ptr<uint8_t[]>* file_buf, si
+            _original_json_doc.type() == simdjson::ondemand::json_type::object) {
+            simdjson::ondemand::object object = _original_json_doc;
+            Status st = JsonFunctions::extract_from_object(object, _parsed_json_root, &_json_value);
+            if (!st.ok()) {
+                fmt::memory_buffer error_msg;
+                fmt::format_to(error_msg, "{}", st.to_string());
+                return return_quality_error(error_msg, std::string((char*)json_str, *size));
+            }
+        } else {
+            _json_value = _original_json_doc;
+        }
+    } catch (simdjson::simdjson_error& e) {
+        fmt::memory_buffer error_msg;
+        fmt::format_to(error_msg, "Encounter error while extract_from_object, error: {}", e.what());
+        return return_quality_error(error_msg, std::string((char*)json_str, *size));
+    }
+
+    if (_json_value.type() == simdjson::ondemand::json_type::array && !_strip_outer_array) {
+        fmt::memory_buffer error_msg;
+        fmt::format_to(error_msg, "{}",
+                       "JSON data is array-object, `strip_outer_array` must be TRUE.");
+        return return_quality_error(error_msg, std::string((char*)json_str, *size));
+    }
+
+    if (_json_value.type() != simdjson::ondemand::json_type::array && _strip_outer_array) {
+        fmt::memory_buffer error_msg;
+        fmt::format_to(error_msg, "{}",
+                       "JSON data is not an array-object, `strip_outer_array` must be FALSE.");
+        return return_quality_error(error_msg, std::string((char*)json_str, *size));
+    }
+    return Status::OK();
+}
+
+Status NewJsonReader::_simdjson_write_columns_by_jsonpath(
+        simdjson::ondemand::object* value, const std::vector<SlotDescriptor*>& slot_descs,
+        std::vector<MutableColumnPtr>& columns, bool* valid) {
+    // write by jsonpath
+    size_t column_num = slot_descs.size();
+    bool has_valid_value = false;
+    size_t cur_row_count = columns[0]->size();
+    for (size_t i = 0; i < column_num; i++) {
+        auto* column_ptr = columns[i].get();
+        simdjson::ondemand::value json_value;
+        Status st;
+        if (i < _parsed_jsonpaths.size()) {
+            st = JsonFunctions::extract_from_object(*value, _parsed_jsonpaths[i], &json_value);
+            if (!st.ok() && !st.is<NOT_FOUND>()) {
+                return st;
+            }
+        }
+        if (i >= _parsed_jsonpaths.size() || st.is<NOT_FOUND>()) {
+            // not match in jsondata.
+            if (!slot_descs[i]->is_nullable()) {
+                RETURN_IF_ERROR(_append_error_msg(
+                        value, "The column `{}` is not nullable, but it's not found in jsondata.",
+                        slot_descs[i]->col_name(), valid));
+                return Status::OK();
+            }
+        } else {
+            RETURN_IF_ERROR(
+                    _simdjson_write_data_to_column(json_value, slot_descs[i], column_ptr, valid));
+            if (!(*valid)) {
```

Comment:
Why return OK if not valid?

Reply:
_append_error_msg will append err msg and continue to process next line